src/battle_script_commands.c

```diff
@@ -8389,6 +8393,13 @@ static bool32 TryDefogClear(u32 battlerAtk, bool32 clear)
         DEFOG_CLEAR(SIDE_STATUS_TOXIC_SPIKES, toxicSpikesAmount, BattleScript_ToxicSpikesDefog, 0);
         DEFOG_CLEAR(SIDE_STATUS_STICKY_WEB, stickyWebAmount, BattleScript_StickyWebDefog, 0);
         DEFOG_CLEAR(SIDE_STATUS_STEELSURGE, steelsurgeAmount, BattleScript_SteelsurgeDefog, 0);
+        if (gBattleWeather & B_WEATHER_FOG)
+        {
+            gBattleWeather &= ~B_WEATHER_FOG;
+            BattleScriptPushCursor();
+            gBattlescriptCurrInstr = BattleScript_FogEnded;
```

Comment:
What happens in vanilla games when there's both a Terrain and Fog when using Defog?
This looks as if Fog will be dispelled but not Terrains.

Reply:
it should clear both but it should be tested just in case

Reply:
From my limited testing in game, Fog gets removed first turn and Terrain gets removed the next time Defog is used.

Reply:
what happens if hazards are up and fog when defog is used?

Reply:
Testing right now. :eyes:

Reply:
Tested myself and it just clears the fog and not the terrain for me. I guess this should be fixed.

Reply:
Sorry not vanilla, but same code! But, it blows away.

![defogblew](https://github.com/rh-hideout/pokeemerald-expansion/assets/75729017/d1215398-6a27-4bdf-a3f2-1e20dc8350b3)